src/ripple/protocol/impl/STLedgerEntry.cpp

```diff
@@ -124,9 +126,18 @@ STLedgerEntry::getJson(JsonOptions options) const
 }
 
 bool
-STLedgerEntry::isThreadedType() const
+STLedgerEntry::isThreadedType(Rules const& rules) const
 {
-    return getFieldIndex(sfPreviousTxnID) != -1;
+    static const std::array<LedgerEntryType, 5> newPreviousTxnIDTypes = {
```

Comment:
this could be `constexpr` instead of `static const` which is an additional benefit of `std::array` . It means the values are ready to use by the compiler front-end during compilation, rather than at the optimization stage. It might not change anything in the end result, but it might also unroll the loop.